code/variables.py

```diff
@@ -0,0 +1,95 @@
+        self.ECCENTRICITY_THRESHOLD_MIN = 0.70 #tending to 0 is a linear e.g. squall line
+        self.OUTER_CLOUD_SHIELD_AREA = 80000.0 #km^2
+        self.INNER_CLOUD_SHIELD_AREA = 30000.0 #km^2
+        self.OUTER_CLOUD_SHIELD_TEMPERATURE = 233 #in K
+        self.INNER_CLOUD_SHIELD_TEMPERATURE = 213 #in K                                                                 
+        self.MINIMUM_DURATION = 6  #min number of frames the MCC must exist for (assuming hrly frames, MCCs is 6hrs) 
+        self.MAXIMUM_DURATION = 24 #max number of framce the MCC can last for   
+        self.DIRS = {'mainDirStr': "/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/output/paperData", 'TRMMdirName':"/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/paperData/TRMM", 'CEoriDirName': "/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/paperData/MERG"}
+        self.startDateTime = "200609110000"
+        self.endDateTime = "200609121200"
+
+
+    def setupJSON(self):
+        data = None;
+        with open('../config.txt') as f:
+            data = json.load(f)
+        self.LATMIN = data['LATMIN']
+        self.LATMAX = data['LATMAX']
+        self.LONMIN = data['LONMIN']
+        self.LONMAX = data['LONMAX']
+        self.XRES = data['XRES']
+        self.YRES = data['YRES']
+        self.TRES = data['TRES']
+        self.LAT_DISTANCE = data['LAT_DISTANCE']
+        self.LON_DISTANCE = data['LON_DISTANCE']
+        self.STRUCTURING_ELEMENT = data['STRUCTURING_ELEMENT']
+        self.T_BB_MAX = data['T_BB_MAX']
+        self.T_BB_MIN = data['T_BB_MIN']
+        self.CONVECTIVE_FRACTION = data['CONVECTIVE_FRACTION']
+        self.MIN_MCS_DURATION = data['MIN_MCS_DURATION']
+        self.AREA_MIN = data['AREA_MIN']
+        self.MIN_OVERLAP = data['MIN_OVERLAP']
+        self.ECCENTRICITY_THRESHOLD_MAX = data['ECCENTRICITY_THRESHOLD_MAX']
+        self.ECCENTRICITY_THRESHOLD_MIN = data['ECCENTRICITY_THRESHOLD_MIN']
+        self.OUTER_CLOUD_SHIELD_AREA = data['OUTER_CLOUD_SHIELD_AREA']
+        self.INNER_CLOUD_SHIELD_AREA = data['INNER_CLOUD_SHIELD_AREA']
+        self.OUTER_CLOUD_SHIELD_TEMPERATURE = data['OUTER_CLOUD_SHIELD_TEMPERATURE']
+        self.INNER_CLOUD_SHIELD_TEMPERATURE = data['INNER_CLOUD_SHIELD_TEMPERATURE']
+        self.MINIMUM_DURATION = data['MINIMUM_DURATION']
+        self.MAXIMUM_DURATION = data['MAXIMUM_DURATION']
+        self.DIRS = data['DIRS']
+        self.startDateTime = data['startDateTime']
+        self.endDateTime = data['endDateTime']
+        
+class GraphVariables(object):
+    def __init__(self):
+        self.edgeWeight = [1,2,3] # weights for the graph edges
+        self.CLOUD_ELEMENT_GRAPH = nx.DiGraph() # graph obj for the CEs meeting criteria
+        self.PRUNED_GRAPH = nx.DiGraph() # graph meeting the CC criteria
+        
+def define_user_variables():
```

Comment:
@youssefbiaz I think function could benefit from a boolean keyword argument like 'useJson' so the caller has the option to use the config.txt file or to just take the defaults that the class would provide.  A keyword argument makes sense because if they don't explicitly ask to use JSON then we can return the class in it's default config.

The way it is written now, the class is instantiated with all the defaults, then the code overwrites all those defaults with values in config.txt.

Something like this would work:

def define_user_variables(useJson=False):
    ''' Function docstring goes here '''
    userVars = UserVariables()
    if useJson:
        userVars.setupJSON()

    return userVars